### PR TITLE
Targeted Quests Rework

### LINKED ISSUE #799: Change "Turn into an Undead" quest target
**Describe your suggestion in full detail below:**

Change "Turn into an Undead" quest target from actual characters to something less specific, since you can get quest about turning character into undead if there is no available characters at your court (quest will be broken) or when you only can do it by tyrannically imprisoning said character, since you can't abduct it.

<details>
<summary>Click to expand</summary>

image

</details>

<details>
<summary>Click to expand</summary>

image
image

</details>

### LINKED ISSUE #817: Can't abduct quest target
**Mod Version**
efdb140

**What expansions do you have installed?**
All

**Please explain your issue in as much detail as possible:**
Can't abduct quest target

**Steps to reproduce the issue:**
???

**Upload an attachment below: .zip of your save, or screenshots:**

<details>
<summary>Click to expand</summary>

image
image

</details>

[Caer_Darrow608_04_24.zip](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/files/3686696/Caer_Darrow608_04_24.zip)

### PR DESCRIPTION
**Pull request by zumbak04**
Fixes #817, closes #799.
## Changelog:
- Target of Mind Control quest no longer is specific character. You should mind control any Smart/Strong/Attractive character.
- Target of Turn into Undead quest no longer is specific character. You should turn any Smart/Strong/Attractive character into undead.

## How to test
Use console to get quests and try to complete them. `event WCQUE.505` to get Turn into Undead quest, `event WCQUE.705` to get Mind Control quest.

### CODE REVIEW

File: events/wc_quest_events.txt [hunk 1]
```diff
@@ -888,6 +882,12 @@ society_quest_event = {
 
 		ai_chance = { factor = 0 }
 	}
+
+	after = {
+		clr_character_flag = control_mind_martial_flag
+		clr_character_flag = control_mind_intrigue_flag
+		clr_character_flag = control_mind_learning_flag
+	}
```
Comment (MemeWizard) [inline events/wc_quest_events.txt]: I would use `clear_flags_with_prefix`, but that's ok too

File: events/wc_quest_events.txt [hunk 2]
```diff
@@ -783,6 +775,12 @@ society_quest_event = {
 
 		ai_chance = { factor = 0 }
 	}
+	
+	after = {
+		clr_character_flag = turn_into_undead_martial_flag
+		clr_character_flag = turn_into_undead_intrigue_flag
+		clr_character_flag = turn_into_undead_learning_flag
+	}
```
Comment (MemeWizard) [inline events/wc_quest_events.txt]: I would use `clear_flags_with_prefix`, but that's ok too

File: localisation/00_zuwc_united_localisation.csv [hunk 3]
```diff
@@ -2984,16 +2975,29 @@ indulgence_ask;The [Root.RelHead.GetTitle] will hear our plea for salvation.;Le
...
-;;;;;;;;;;;;;;x
-;;;;;;;;;;;;;;x
-;;;;;;;;;;;;;;x
-;;;;;;;;;;;;;;x
+quest_control_mind_martial;Mind Control Commander;;;;;;;;;;;;;x
+quest_control_mind_intrigue;Mind Control Schemer;;;;;;;;;;;;;x
+quest_control_mind_learning;Mind Control Scholar;;;;;;;;;;;;;x
+quest_control_mind_martial_desc;The Cult needs all sort of servants. For some missions, a true cultist is needed. Other and more menial tasks, however, do not require the same kind of power and wit. They demand a loyal commander to carry out the cult�s wishes.;;;;;;;;;;;;;x
+quest_control_mind_intrigue_desc;The Cult needs all sort of servants. For some missions, a true cultist is needed. Other and more menial tasks, however, do not require the same kind of power and wit. They demand a loyal schemer to carry out the cult�s wishes.;;;;;;;;;;;;;x
+quest_control_mind_learning_desc;The Cult needs all sort of servants. For some missions, a true cultist is needed. Other and more menial tasks, however, do not require the same kind of power and wit. They demand a loyal scholar to carry out the cult�s wishes.;;;;;;;;;;;;;x
```
[the remark this answers is not in the document]
Comment (zumbak04) [reply localisation/00_zuwc_united_localisation.csv]: Fixed!

File: localisation/00_zuwc_united_localisation.csv [hunk 4]
```diff
@@ -2984,16 +2975,29 @@ indulgence_ask;The [Root.RelHead.GetTitle] will hear our plea for salvation.;Le
...
+quest_turn_into_undead_intrigue_desc;Our almighty Lich King needs all sort of servants. The Cult believes that the dead are the most faithful servants. Turn a promising schemer into the undead to carry out the cult�'s wishes.;;;;;;;;;;;;;x
+quest_turn_into_undead_learning_desc;Our almighty Lich King needs all sort of servants. The Cult believes that the dead are the most faithful servants. Turn a promising scholar into the undead to carry out the cult�'s wishes.;;;;;;;;;;;;;x
+quest_turn_into_undead_martial_tooltip;To complete this mission, imprison a character who has at least �Y15 Martial�!, �Gright-click�! on the portrait and select �Y'Turn Undead'�!.;;;;;;;;;;;;;x
+quest_turn_into_undead_intrigue_tooltip;To complete this mission, imprison a character who has at least �Y15 Intrigue�!, �Gright-click�! on the portrait and select �Y'Turn Undead'�!.;;;;;;;;;;;;;x
+quest_turn_into_undead_learning_tooltip;To complete this mission, imprison a character who has at least �Y15 Learning�!, �Gright-click�! on the portrait and select �Y'Turn Undead'�!.;;;;;;;;;;;;;x
```
[the remark this answers is not in the document]
Comment (zumbak04) [reply localisation/00_zuwc_united_localisation.csv]: Fixed!

File: localisation/00_zuwc_united_localisation.csv [hunk 5]
```diff
@@ -2984,16 +2975,29 @@ indulgence_ask;The [Root.RelHead.GetTitle] will hear our plea for salvation.;Le
...
+quest_control_mind_learning_desc;The Cult needs all sort of servants. For some missions, a true cultist is needed. Other and more menial tasks, however, do not require the same kind of power and wit. They demand a loyal scholar to carry out the cult�s wishes.;;;;;;;;;;;;;x
+quest_control_mind_martial_tooltip;To complete this mission, �Gright-click�! on the portrait of a character who has at least �Y15 Martial�!, select �Y'Mind control'�! and successfully enthrall the character.;;;;;;;;;;;;;x
+quest_control_mind_intrigue_tooltip;To complete this mission, �Gright-click�! on the portrait of a character who has at least �Y15 Intrigue�!, select �Y'Mind control'�! and successfully enthrall the character.;;;;;;;;;;;;;x
+quest_control_mind_learning_tooltip;To complete this mission, �Gright-click�! on the portrait of a character who has at least �Y15 Learning�!, select �Y'Mind control'�! and successfully enthrall the character.;;;;;;;;;;;;;x
```
[the remark this answers is not in the document]
Comment (zumbak04) [reply localisation/00_zuwc_united_localisation.csv]: Fixed!

File: events/wc_quest_events.txt (hunk 1 above)
Comment (zumbak04) [reply events/wc_quest_events.txt]: There may be other flags in the future starting with `control_mind_` or even `quest_control_mind_`. I wouldn't do it.

File: common/scripted_triggers/wc_society_scripted_triggers.txt [hunk 6]
```diff
@@ -271,4 +271,69 @@ sisterhood_of_elune_potential_trigger = {
...
+# Mind Control
+quest_control_mind_martial_target_trigger = {
+	martial = 15
+}
+quest_control_mind_intrigue_target_trigger = {
+	intrigue = 15
+}
+quest_control_mind_learning_target_trigger = {
+	learning = 15
+}
```
[the remark this answers is not in the document]
Comment (zumbak04) [reply common/scripted_triggers/wc_society_scripted_triggers.txt]: Quests targets were changed in https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/pull/870/commits/6e12bb92e0c70a391027bd61564aeacfedad8af9.

File: common/scripted_triggers/wc_society_scripted_triggers.txt [hunk 7]
```diff
@@ -271,4 +271,69 @@ sisterhood_of_elune_potential_trigger = {
...
+# Turn into Undead
+quest_turn_into_undead_martial_target_trigger = {
+	martial = 15
+}
+quest_turn_into_undead_intrigue_target_trigger = {
+	intrigue = 15
+}
+quest_turn_into_undead_learning_target_trigger = {
+	learning = 15
+}
```
[the remark this answers is not in the document]
Comment (zumbak04) [reply common/scripted_triggers/wc_society_scripted_triggers.txt]: Quests targets were changed in https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/pull/870/commits/6e12bb92e0c70a391027bd61564aeacfedad8af9.

File: localisation/00_zuwc_united_localisation.csv [hunk 8]
```diff
@@ -2985,15 +2976,29 @@ repentance_ask;The [Root.RelHead.GetTitle] will hear our plea for salvation.;Le
...
-;;;;;;;;;;;;;;x
-;;;;;;;;;;;;;;x
-;;;;;;;;;;;;;;x
-;;;;;;;;;;;;;;x
-;;;;;;;;;;;;;;x
-;;;;;;;;;;;;;;x
-;;;;;;;;;;;;;;x
+quest_control_mind_smart;Mind Control Clever Man or Woman;;;;;;;;;;;;;x
+quest_control_mind_strong;Mind Control Sturdy Man or Woman;;;;;;;;;;;;;x
+quest_control_mind_fair;Mind Control Attractive Man or Woman;;;;;;;;;;;;;x
```
Comment (DukeJeans) [inline localisation/00_zuwc_united_localisation.csv]: I'd use "Individual" e.g. "Clever Individual" to maintain your need for narrative sense

Comment (zumbak04) [reply localisation/00_zuwc_united_localisation.csv]: Implemented!

**Comment by MemeWizard** [comment]
You can mind control/turn into undead as many smart/strong/attractive character as you like, as long you have this quest enabled, because casting appropriate spell on quest target is free.

<details>
<summary>Click to expand</summary>

image
image

</details>

**Comment by zumbak04** [comment]
> You can mind control/turn into undead as many smart/strong/attractive character as you like, as long you have this quest enabled, because casting appropriate spell on quest target is free.

@MemeWizard it should be free to avoid problems like the quest reward doesn't cover the cost of the decisions to complete this quest.
I see 2 ways to solve this:
1) Block abusing itself. Mind Control and Turn Undead decisions set flags that make the decisions paid.
2) Block Mind Control and Turn Undead decisions if they're ongoing.

**Comment by MemeWizard** [comment]
> Block Mind Control and Turn Undead decisions if they're ongoing.

Sounds like the best option

**Comment by MemeWizard** [comment]
Abomination don't work correctly - you don't get society currency, moreover, you pay 100.

<details>
<summary>Click to expand</summary>

image
image

</details>

**Comment by zumbak04** [comment]
> Abomination don't work correctly - you don't get society currency, moreover, you pay 100.

Fixed!

**Pull request merged**